lib/taskana-core/src/main/java/pro/taskana/impl/WorkbasketServiceImpl.java

```diff
@@ -779,19 +782,67 @@ private void markWorkbasketForDeletion(String workbasketId)
             WorkbasketImpl workbasket = workbasketMapper.findById(workbasketId);
             workbasket.setMarkedForDeletion(true);
             workbasketMapper.update(workbasket);
-            distributionTargetMapper.deleteAllDistributionTargetsBySourceId(workbasketId);
-            distributionTargetMapper.deleteAllDistributionTargetsByTargetId(workbasketId);
-            workbasketAccessMapper.deleteAllAccessItemsForWorkbasketId(workbasketId);
+            deleteReferencesToWorkbasket(workbasketId);
         } finally {
             taskanaEngine.returnConnection();
             LOGGER.debug("exit from markWorkbasketForDeletion(workbasketId = {}).", workbasketId);
         }
     }
 
+    public BulkOperationResults<String, TaskanaException> deleteWorkbaskets(List<String> workbasketsIds)
+        throws NotAuthorizedException, InvalidArgumentException {
+        LOGGER.debug("entry to deleteWorkbaskets(workbasketId = {})", LoggerUtils.listToString(workbasketsIds));
+
+        taskanaEngine.checkRoleMembership(TaskanaRole.BUSINESS_ADMIN, TaskanaRole.ADMIN);
+
+        try {
+            taskanaEngine.openConnection();
+            if (workbasketsIds == null || workbasketsIds.isEmpty()) {
+                throw new InvalidArgumentException("List of WorkbasketIds must not be null.");
+            }
+            WorkbasketQuery workbasketQuery = createWorkbasketQuery();
+
+            List<String> existingWorkbasketsIds = workbasketQuery.idIn(workbasketsIds.stream().toArray(String[]::new))
```

Comment:
Actually we don't even need this call. deleteWorkbasket will find out if the workbasket exists. If it doesn't the error will be stored in the bulk log. 
If we do this call, we will read the entire workbasket table even if we just want to delete 2 workbaskets.

Reply:
Yes, you are right. I will remove it

Reply:
You are right, Now it's deleted